charms/knative-eventing/src/lightkube_custom_resources/operator.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Knative Eventing custom resource classes."""
+
+from lightkube.generic_resource import create_namespaced_resource
+
+# Knative Operator's KnativeEventing CRDs
+# version 1.1.x
+KnativeEventing_v1alpha1 = create_namespaced_resource(
+    group="operator.knative.dev",
+    version="v1alpha1",
+    kind="KnativeEventing",
+    plural="knativeeventings",
+    verbs=None,
+)
```

Comment:
(I realise I'm to blame for this, but now I think I made a mistake before!)
I think by putting these directly in the main scope of this file, this namespaced resource gets created anytime anyone imports something from this file.  For example, all of these are equivalent:
```python
from lightkube_custom_resources.operator import KnativeEventing_v1alpha1
import lightkube_custom_resources.operator
from lightkube_custom_resources.operator import SomethingCompletelyUnrelated
```

I wonder if these should be behind functions, like:
```python
def create_KnativeEventing_v1alpha1():
  return create_namespaced_resource(...)
```

and used like:
```python
from lightkube_custom_resources.operator import create_KnativeEventing_v1alpha1

crd_class = create_KnativeEventing_v1alpha1()
```

This way it only gets called when someone actually means to.

Reply:
> (I realise I'm to blame for this, but now I think I made a mistake before!)
I think by putting these directly in the main scope of this file, this namespaced resource gets created anytime anyone imports something from this file.

Is this behaviour something completely undesired, though? We'd have access to different classes, anyway.